Content.Client/GameObjects/Components/Power/SolarControlConsoleBoundUserInterface.cs

```diff
@@ -0,0 +1,233 @@
+using System;
+using Content.Client.UserInterface;
+using Content.Client.UserInterface.Stylesheets;
+using Content.Shared.GameObjects.Components.Power;
+using Robust.Client.GameObjects.Components.UserInterface;
+using Robust.Client.Graphics.Drawing;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Shared.GameObjects.Components.UserInterface;
+using Robust.Shared.Interfaces.Timing;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+
+namespace Content.Client.GameObjects.Components.Power
+{
+    public class SolarControlConsoleBoundUserInterface : BoundUserInterface
+    {
+        [Dependency]
+        private IGameTiming _gameTiming;
+
+        private SolarControlWindow _window;
+        private SolarControlConsoleBoundInterfaceState _lastState = new SolarControlConsoleBoundInterfaceState(0, 0, 0, 0);
+
+        protected override void Open()
+        {
+            base.Open();
+
+            _window = new SolarControlWindow(_gameTiming);
+            _window.OnClose += Close;
+            _window.PanelRotation.OnTextEntered += (text) => {
+                double value;
+                if (double.TryParse(text.Text, out value))
+                {
+                    SolarControlConsoleAdjustMessage msg = new SolarControlConsoleAdjustMessage();
+                    msg.Rotation = Angle.FromDegrees(value);
+                    msg.AngularVelocity = _lastState.AngularVelocity;
+                    SendMessage(msg);
+                }
+            };
+            _window.PanelVelocity.OnTextEntered += (text) => {
+                double value;
+                if (double.TryParse(text.Text, out value))
+                {
+                    SolarControlConsoleAdjustMessage msg = new SolarControlConsoleAdjustMessage();
+                    msg.Rotation = _lastState.Rotation;
+                    msg.AngularVelocity = Angle.FromDegrees(value / 60);
+                    SendMessage(msg);
+                }
+            };
+            _window.OpenCenteredMinSize();
+        }
+
+        public SolarControlConsoleBoundUserInterface(ClientUserInterfaceComponent owner, object uiKey) : base(owner, uiKey)
+        {
+        }
+
+        private string FormatAngle(Angle d)
+        {
+            return (Math.Floor(d.Degrees * 10) / 10).ToString();
```

Comment:
Would you be able to do this with a [standard numeric formatter](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings)? As of now, it's quite hard to understand what effect this has on the format.

Reply:
Probably yes (format string "F1" seems like it'd do it) but that requires standard library knowledge I didn't have when I wrote that.